orangecontrib/prototypes/widgets/owinteractions.py

```diff
@@ -113,28 +219,41 @@ def compute_score(self, state):
 		removed = self.interaction.removed_h[attr1, attr2] / h
 		return -score, score, gain1, gain2, removed
 
+	def _update_model(self):
```

Comment:
I'd avoid the reimplementation here. If your visualizer does not used `gui.TableBarItem.BarRole`, no change is needed.

If you use the role (which I think you should not), then you can stop it being reset by just reimplementing `bar_length()` to return None.

Reply:
You are right, I think i changed it before I settled on a final solution and now I no longer need it, so I will remove it completely.

What should I use to get the scores to the delegat instead of Roles?

Reply:
Roles are fine, but if you want to avoid the effects of that setData statement (that is in VizrankAttrPair but not in your code), just don't use `gui.TableBarItem.BarRole`. Any other Role would work.